### PR TITLE
Add core package with data model and JSON schema

### PR DESCRIPTION
**Pull request by dorranh**
This PR adds a new package called `core` which defines our core data model used to represent the dependency tree extracted from nixpkgs. This can be used by the utility in #42 for constructing its output.

As discussed with @GuillaumeDesforges, I've checked the generated JSON schema into the repo for downstream users to use. We will want to add a CI step checking that it is up to date in the future.

### CODE REVIEW

File: core/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# core
+
+Core data model used for extracting Nix dependencies and related utilities.
+
+To generate a JSON schema you can run the following while in the repositories
```
Comment (GuillaumeDesforges) [inline core/README.md:5]:
```suggestion
To generate a JSON schema you can run the following while in the repository
```

File: core/explorer/core/__main__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from explorer.core import model
+
+import click
+
+
+@click.command()
+@click.option(
+    "--out",
```
Comment (GuillaumeDesforges) [inline core/explorer/core/__main__.py:8]: what if I want to output to stdout?

File: core/explorer/core/model.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class BuildInput(BaseModel):
+    """A build input to a Nix derivation"""
+
+    build_input_type: BuildInputType = Field(description="The type of build input.")
+    package: "Package" = Field("The input package")
+
+
+class Package(BaseModel):
+    """A Nix package"""
```
Comment (GuillaumeDesforges) [inline core/explorer/core/model.py:45]:
```suggestion
    """A Nix package, which is an evaluated (not realized) derivation."""
```

**Review by GuillaumeDesforges — APPROVED** [review]
lgtm, comments are nitpicky

File: core/explorer/core/__main__.py (hunk 2 above)
Comment (dorranh) [reply core/explorer/core/__main__.py:8]: Then you're out of luck ;) 

(Added support here: [3faa2d0](https://github.com/tweag/nixpkgs-graph-explorer/pull/56/commits/3faa2d00f3e504a08dc5399a3f894325419a1921))

**Pull request merged**